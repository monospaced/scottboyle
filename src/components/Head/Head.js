import React from "react";

import "../../assets/android-chrome-192x192.png";
import "../../assets/android-chrome-512x512.png";
import "../../assets/apple-touch-icon.png";
import "../../assets/browserconfig.xml";
import "../../assets/favicon.ico";
import "../../assets/favicon-16x16.png";
import "../../assets/favicon-32x32.png";
import "../../assets/manifest.webmanifest";
import "../../assets/mstile-150x150.png";
import "../../assets/safari-pinned-tab.svg";
import vars from "!css-variables-loader!../../theme/vars.css";
import { version } from "../../../package.json";

const themeColor = vars["--theme-color"];

const Head = ({ meta }) => {
  return (
    <head>
      <meta charSet="utf-8" />
      {meta}
      <meta name="msapplication-TileColor" content={themeColor} />
      <meta name="theme-color" content={themeColor} />
      <meta name="viewport" content="width=device-width" />
      <link
        rel="apple-touch-icon"
        sizes="180x180"
        href="/apple-touch-icon.png"
      />
      <link
        rel="icon"
        type="image/png"
        sizes="32x32"
        href="/favicon-32x32.png"
      />
      <link
        rel="icon"
        type="image/png"
        sizes="16x16"
        href="/favicon-16x16.png"
      />
      <link rel="manifest" href="/manifest.webmanifest" />
      <link rel="mask-icon" href="/safari-pinned-tab.svg" color={themeColor} />
      <link rel="stylesheet" href={`/styles.css?v=${version}`} />
    </head>
  );
};

export default Head;
