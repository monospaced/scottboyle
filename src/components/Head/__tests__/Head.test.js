import React from "react";
import { render } from "enzyme";

import data from "../../../scripts/__mocks__/data.js";
import Head from "../Head.js";

jest.mock("../../../../package.json", () => ({ version: "0.0.0" }));

describe("Head component", () => {
  it("should render correctly", () => {
    const component = render(<Head />);
    expect(component).toMatchSnapshot();
  });
});
