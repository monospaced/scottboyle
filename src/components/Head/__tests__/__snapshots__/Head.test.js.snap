// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head component should render correctly 1`] = `
Array [
  <meta
    charset="utf-8"
  />,
  <meta
    content="hotpink"
    name="msapplication-TileColor"
  />,
  <meta
    content="hotpink"
    name="theme-color"
  />,
  <meta
    content="width=device-width"
    name="viewport"
  />,
  <link
    href="/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    color="hotpink"
    href="/safari-pinned-tab.svg"
    rel="mask-icon"
  />,
  <link
    href="/styles.css?v=0.0.0"
    rel="stylesheet"
  />,
]
`;
